### PR TITLE
Add API endpoint for getting job workflow tree

### PR DESCRIPTION
**Pull request by ProTip**
* Adds `job/[ID]/workflow` endpoint to public API for retrieving a job's workflow tree

This is preparatory work for moving more functionality out into components in the `grails-spa` project. Specifically it is being utilized by the new execution output viewer component.

**Docs PR**
rundeck/docs#338

### CODE REVIEW

**Comment by gschueler** [comment]
it would "work", but needs some more changes to be a full API endpoint:
* api version check,
* error responses should be use apiService for correct response format
* if only JSON is supported, should respond 405 for other Accept requests.

Ideally, the workflowJson would just be updated to be a full api endpoint, and usage of workflowJson in the client-side js could be changed to use a true API request

**Comment by ProTip** [comment]
@gschueler Converted to API endpoint and the usage I found converted to utilize it via an update `ts-rundeck` client.

Will merge the `ui-trellis` branch into master and update commit ref before merging this one.

File: rundeckapp/grails-app/controllers/rundeck/controllers/ScheduledExecutionController.groovy [hunk 1]
```diff
@@ -627,7 +639,10 @@ class ScheduledExecutionController  extends ControllerBase{
                 ),
                 AuthConstants.ACTION_VIEW, 'Job', params.id
         )) {
-            return
+            return apiService.renderErrorFormat(response, [status: HttpServletResponse.SC_FORBIDDEN,
+                                                           code  : 'api.error.item.unauthorized', args: ['Run', 'Job ' +
```
Comment (gschueler) [inline rundeckapp/grails-app/controllers/rundeck/controllers/ScheduledExecutionController.groovy]:
```suggestion
                                                           code  : 'api.error.item.unauthorized', args: ['View', 'Job ' +
```

**Comment by ProTip** [comment]
I realized when adding the docs that API v33 isn't the pending bump I thought it was.

**Pull request merged**